src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -129,13 +129,13 @@ public virtual string CreateToken(SecurityTokenDescriptor tokenDescriptor)
             if (tokenDescriptor == null)
                 throw LogHelper.LogArgumentNullException(nameof(tokenDescriptor));
 
-            if (tokenDescriptor.Payload == null || !tokenDescriptor.Payload.Any())
-                throw LogHelper.LogArgumentNullException(nameof(tokenDescriptor.Payload));
+            if (tokenDescriptor.Claims == null || !tokenDescriptor.Claims.Any())
+                throw LogHelper.LogArgumentNullException(nameof(tokenDescriptor.Claims));
 
             if (tokenDescriptor.SigningCredentials == null)
                 throw LogHelper.LogArgumentNullException(nameof(tokenDescriptor.SigningCredentials));
 
-            var payload = new Dictionary<string, object>(tokenDescriptor.Payload);
+            var payload = new Dictionary<string, object>(tokenDescriptor.Claims);
```

Comment:
Can we open a bug to 'remember' what the actual audience should be and then we don't have to copy the dictionary.